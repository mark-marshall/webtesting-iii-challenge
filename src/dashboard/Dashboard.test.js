// ### Dashboard

// - shows the controls and display (Done)

import React from 'react';
import * as rt from 'react-testing-library';

import Dashboard from './Dashboard';

afterEach(rt.cleanup);

describe('<Dashboard />', () => {
    it('renders the display', () => {
        const wrap = rt.render(<Dashboard />);
        expect(wrap.getByTestId(/displayDiv/i));
        expect(wrap.asFragment()).toMatchSnapshot();
    })
    it('renders the controls', () => {
        const wrap = rt.render(<Dashboard />);
        expect(wrap.getByTestId(/displayDiv/i));
        expect(wrap.asFragment()).toMatchSnapshot();
    })
})
