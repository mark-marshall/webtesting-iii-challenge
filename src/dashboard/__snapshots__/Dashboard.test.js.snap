// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> renders the controls 1`] = `
<DocumentFragment>
  <div
    class="display panel"
    data-testid="displayDiv"
  >
    <div
      class="led green-led"
      data-testid="lockedDiv"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="closedDiv"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
    data-testid="controlDiv"
  >
    <button
      class="toggle-btn"
      data-testid="lockedButton"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="closedButton"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;

exports[`<Dashboard /> renders the display 1`] = `
<DocumentFragment>
  <div
    class="display panel"
    data-testid="displayDiv"
  >
    <div
      class="led green-led"
      data-testid="lockedDiv"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="closedDiv"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
    data-testid="controlDiv"
  >
    <button
      class="toggle-btn"
      data-testid="lockedButton"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="closedButton"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
